Socket Timeout when running Flank 4.3.1-Snapshot

When running the current snapshot version we are very often having failed builds with the following exception: 

```
Request failed, retrying 1x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 2x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 3x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 4x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 5x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 6x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 7x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 8x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 9x java.net.SocketTimeoutException: connect timed out
Request failed, retrying 10x java.net.SocketTimeoutException: connect timed out

Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.android.AndroidRunCommand@5f9be66c): java.io.IOException: Request failed
	at picocli.CommandLine.execute(CommandLine.java:1168)
	at picocli.CommandLine.access$800(CommandLine.java:141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at picocli.CommandLine.run(CommandLine.java:1974)
	at picocli.CommandLine.run(CommandLine.java:1920)
	at ftl.Main$Companion.main(Main.kt:48)
	at ftl.Main.main(Main.kt)
Caused by: java.io.IOException: Request failed
	at ftl.http.ExecuteWithRetryKt.executeWithRetry(ExecuteWithRetry.kt:20)
	at ftl.gc.GcToolResults.getDefaultBucket(GcToolResults.kt:79)
	at ftl.args.yml.GcloudYmlParams.<init>(GcloudYml.kt:52)
	at ftl.args.yml.GcloudYmlParams.<init>(GcloudYml.kt:37)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41)
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflect_api(KCallableImpl.kt:166)
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:110)
	at com.fasterxml.jackson.module.kotlin.KotlinValueInstantiator.createFromObjectWith(KotlinValueInstantiator.kt:135)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:195)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:488)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:530)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:528)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:417)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3004)
	at ftl.args.AndroidArgs$Companion.load(AndroidArgs.kt:188)
	at ftl.args.AndroidArgs$Companion.load(AndroidArgs.kt:184)
	at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:33)
	at picocli.CommandLine.execute(CommandLine.java:1160)
	... 9 more
Caused by: java.net.SocketTimeoutException: connect timed out
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1334)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1309)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:259)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:113)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:84)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1011)
	at com.google.api.client.auth.oauth2.TokenRequest.executeUnparsed(TokenRequest.java:322)
	at com.google.api.client.auth.oauth2.TokenRequest.execute(TokenRequest.java:346)
	at com.google.api.client.googleapis.auth.oauth2.GoogleCredential.executeRefreshToken(GoogleCredential.java:394)
	at com.google.api.client.auth.oauth2.Credential.refreshToken(Credential.java:494)
	at com.google.api.client.auth.oauth2.Credential.intercept(Credential.java:217)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:897)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:499)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:432)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:549)
	at ftl.http.ExecuteWithRetryKt.executeWithRetry(ExecuteWithRetry.kt:13)
	... 37 more
```

With the above change I start getting the following instead. 

I am going to double check that we don't have connectivity issues in our CI environment, but this doesn't happen 100% of the time so it would be odd if we did. 

```
Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.android.AndroidRunCommand@363042d7): com.google.cloud.storage.StorageException: Error getting access token for service account: connect timed out
	at picocli.CommandLine.execute(CommandLine.java:1168)
	at picocli.CommandLine.access$800(CommandLine.java:141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at picocli.CommandLine.run(CommandLine.java:1974)
	at picocli.CommandLine.run(CommandLine.java:1920)
	at ftl.Main$Companion.main(Main.kt:48)
	at ftl.Main.main(Main.kt)
Caused by: com.google.cloud.storage.StorageException: Error getting access token for service account: connect timed out
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:226)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.list(HttpStorageRpc.java:332)
	at com.google.cloud.storage.StorageImpl$6.call(StorageImpl.java:303)
	at com.google.cloud.storage.StorageImpl$6.call(StorageImpl.java:299)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.StorageImpl.listBuckets(StorageImpl.java:297)
	at com.google.cloud.storage.StorageImpl.list(StorageImpl.java:285)
	at ftl.args.ArgsHelper.createGcsBucket(ArgsHelper.kt:119)
	at ftl.args.AndroidArgs.<init>(AndroidArgs.kt:92)
	at ftl.args.AndroidArgs$Companion.load(AndroidArgs.kt:191)
	at ftl.args.AndroidArgs$Companion.load(AndroidArgs.kt:184)
	at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:33)
	at picocli.CommandLine.execute(CommandLine.java:1160)
	... 9 more
Caused by: java.io.IOException: Error getting access token for service account: connect timed out
	at com.google.auth.oauth2.ServiceAccountCredentials.refreshAccessToken(ServiceAccountCredentials.java:401)
	at com.google.auth.oauth2.OAuth2Credentials.refresh(OAuth2Credentials.java:181)
	at com.google.auth.oauth2.OAuth2Credentials.getRequestMetadata(OAuth2Credentials.java:167)
	at com.google.auth.http.HttpCredentialsAdapter.initialize(HttpCredentialsAdapter.java:96)
	at com.google.cloud.http.HttpTransportOptions$1.initialize(HttpTransportOptions.java:159)
	at com.google.cloud.http.CensusHttpModule$CensusHttpRequestInitializer.initialize(CensusHttpModule.java:109)
	at com.google.api.client.http.HttpRequestFactory.buildRequest(HttpRequestFactory.java:93)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.buildHttpRequest(AbstractGoogleClientRequest.java:381)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:499)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:432)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:549)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.list(HttpStorageRpc.java:328)
	... 22 more
Caused by: java.net.SocketTimeoutException: connect timed out
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1334)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1309)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:259)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:113)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:84)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1011)
	at com.google.auth.oauth2.ServiceAccountCredentials.refreshAccessToken(ServiceAccountCredentials.java:399)
	... 33 more
```

--- a/test_runner/build.gradle.kts
+++ b/test_runner/build.gradle.kts
@@ -114,8 +114,9 @@ dependencies {
     compile("com.fasterxml.jackson.dataformat:jackson-dataformat-xml:${Versions.JACKSON}")
     compile("com.fasterxml.woodstox:woodstox-core:5.1.0")
 
-    // https://github.com/googleapis/google-oauth-java-client
-    compile("com.google.oauth-client:google-oauth-client-jetty:1.23.0")
+    // End of Life https://github.com/googleapis/google-oauth-java-client
+    // Use https://github.com/googleapis/google-auth-library-java instead
+    compile("com.google.auth:google-auth-library-oauth2-http:0.13.0")
 
     // https://github.com/jhy/jsoup/releases
     testImplementation("org.jsoup:jsoup:1.11.3")

--- a/test_runner/src/main/kotlin/ftl/cli/auth/LoginCommand.kt
+++ b/test_runner/src/main/kotlin/ftl/cli/auth/LoginCommand.kt
@@ -1,6 +1,5 @@
 package ftl.cli.auth
 
-import ftl.gc.GcAuth
 import picocli.CommandLine
 
 @CommandLine.Command(
@@ -16,8 +15,6 @@ import picocli.CommandLine
 )
 class LoginCommand : Runnable {
     override fun run() {
-        GcAuth.authorizeUser()
-        println("Saving credential to ${GcAuth.CRED.absolutePath}")
     }
 
     @CommandLine.Option(

--- a/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
+++ b/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
@@ -57,7 +57,11 @@ object FtlConstants {
                     .build()
             )
         } else {
-            // https://github.com/googleapis/google-auth-library-java
+            // Authenticate with https://github.com/googleapis/google-auth-library-java
+            // Scope is required.
+            // https://developers.google.com/identity/protocols/googlescopes
+            // https://developers.google.com/identity/protocols/application-default-credentials
+            // https://cloud.google.com/sdk/gcloud/reference/alpha/compute/instances/set-scopes
             HttpCredentialsAdapter(
                 ServiceAccountCredentials.getApplicationDefault()
                     .createScoped(listOf("https://www.googleapis.com/auth/cloud-platform"))

--- a/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
+++ b/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
@@ -7,8 +7,6 @@ import com.google.api.client.http.HttpRequestInitializer
 import com.google.api.client.http.javanet.NetHttpTransport
 import com.google.api.client.json.JsonFactory
 import com.google.auth.http.HttpCredentialsAdapter
-import com.google.auth.oauth2.AccessToken
-import com.google.auth.oauth2.GoogleCredentials
 import com.google.auth.oauth2.ServiceAccountCredentials
 import ftl.args.AndroidArgs
 import ftl.args.IArgs
@@ -16,7 +14,6 @@ import ftl.args.IosArgs
 import ftl.http.TimeoutHttpRequestInitializer
 import java.nio.file.Path
 import java.nio.file.Paths
-import java.util.Date
 
 object FtlConstants {
     var useMock = false

--- a/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
+++ b/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
@@ -1,15 +1,19 @@
 package ftl.config
 
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.googleapis.testing.auth.oauth2.MockGoogleCredential
 import com.google.api.client.googleapis.util.Utils
+import com.google.api.client.http.HttpRequestInitializer
 import com.google.api.client.http.javanet.NetHttpTransport
 import com.google.api.client.json.JsonFactory
 import com.google.auth.http.HttpCredentialsAdapter
 import com.google.auth.oauth2.AccessToken
 import com.google.auth.oauth2.GoogleCredentials
+import com.google.auth.oauth2.ServiceAccountCredentials
 import ftl.args.AndroidArgs
 import ftl.args.IArgs
 import ftl.args.IosArgs
+import ftl.http.TimeoutHttpRequestInitializer
 import java.nio.file.Path
 import java.nio.file.Paths
 import java.util.Date
@@ -45,15 +49,20 @@ object FtlConstants {
         Paths.get(System.getProperty("user.home"), ".config/gcloud/application_default_credentials.json")
     }
 
-    val credential by lazy {
-        val cred = if (useMock) {
-            GoogleCredentials.newBuilder().setAccessToken(AccessToken("ss", Date(9999))).build()
+    val credential: HttpRequestInitializer by lazy {
+        if (useMock) {
+            TimeoutHttpRequestInitializer(
+                MockGoogleCredential.Builder()
+                    .setTransport(MockGoogleCredential.newMockHttpTransportWithSampleTokenResponse())
+                    .build()
+            )
         } else {
-            GoogleCredentials.getApplicationDefault()
+            // https://github.com/googleapis/google-auth-library-java
+            HttpCredentialsAdapter(
+                ServiceAccountCredentials.getApplicationDefault()
+                    .createScoped(listOf("https://www.googleapis.com/auth/cloud-platform"))
+            )
         }
-
-        // https://github.com/googleapis/google-auth-library-java
-        HttpCredentialsAdapter(cred)
     }
 
     const val localResultsDir = "results"

--- a/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
+++ b/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt
@@ -1,21 +1,18 @@
 package ftl.config
 
-import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
-import com.google.api.client.googleapis.testing.auth.oauth2.MockGoogleCredential
 import com.google.api.client.googleapis.util.Utils
-import com.google.api.client.http.HttpRequestInitializer
 import com.google.api.client.http.javanet.NetHttpTransport
 import com.google.api.client.json.JsonFactory
+import com.google.auth.http.HttpCredentialsAdapter
 import com.google.auth.oauth2.AccessToken
 import com.google.auth.oauth2.GoogleCredentials
 import ftl.args.AndroidArgs
 import ftl.args.IArgs
 import ftl.args.IosArgs
-import ftl.gc.GcAuth
-import ftl.http.TimeoutHttpRequestInitializer
 import java.nio.file.Path
 import java.nio.file.Paths
+import java.util.Date
 
 object FtlConstants {
     var useMock = false
@@ -48,35 +45,15 @@ object FtlConstants {
         Paths.get(System.getProperty("user.home"), ".config/gcloud/application_default_credentials.json")
     }
 
-    private val credentialObj: GoogleCredential by lazy {
-        try {
-            // Scope is required.
-            // https://developers.google.com/identity/protocols/googlescopes
-            // https://developers.google.com/identity/protocols/application-default-credentials
-            // https://cloud.google.com/sdk/gcloud/reference/alpha/compute/instances/set-scopes
-            return@lazy if (useMock) {
-                MockGoogleCredential.Builder()
-                    .setTransport(MockGoogleCredential.newMockHttpTransportWithSampleTokenResponse())
-                    .build()
-            } else {
-                if (GcAuth.hasUserAuth()) {
-                    GcAuth.authorizeUser()
-                } else {
-                    GoogleCredential.getApplicationDefault()
-                        .createScoped(listOf("https://www.googleapis.com/auth/cloud-platform"))
-                }
-            }
-        } catch (e: Exception) {
-            throw RuntimeException(e)
+    val credential by lazy {
+        val cred = if (useMock) {
+            GoogleCredentials.newBuilder().setAccessToken(AccessToken("ss", Date(9999))).build()
+        } else {
+            GoogleCredentials.getApplicationDefault()
         }
-    }
 
-    val credential: HttpRequestInitializer by lazy {
-        return@lazy TimeoutHttpRequestInitializer(credentialObj)
-    }
-
-    val googleCredentials: GoogleCredentials by lazy {
-        GoogleCredentials.create(AccessToken(credentialObj.accessToken, null))
+        // https://github.com/googleapis/google-auth-library-java
+        HttpCredentialsAdapter(cred)
     }
 
     const val localResultsDir = "results"

--- a/test_runner/src/main/kotlin/ftl/gc/GcAuth.kt
+++ b/test_runner/src/main/kotlin/ftl/gc/GcAuth.kt
@@ -1,17 +1,12 @@
 package ftl.gc
 
-import com.google.api.client.auth.oauth2.Credential
 import com.google.api.client.auth.oauth2.StoredCredential
-import com.google.api.client.extensions.java6.auth.oauth2.AuthorizationCodeInstalledApp
-import com.google.api.client.extensions.jetty.auth.oauth2.LocalServerReceiver
 import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow
 import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
 import com.google.api.client.json.jackson2.JacksonFactory
 import com.google.api.client.util.store.FileDataStoreFactory
-import ftl.config.FtlConstants
 import java.io.File
-import java.io.IOException
 
 // https://github.com/googleapis/google-oauth-java-client
 // GoogleAuthorizationCodeFlow usage based on https://developers.google.com/sheets/api/quickstart/java
@@ -52,23 +47,11 @@ object GcAuth {
             .setAccessToken(accessToken)
     }
 
-    private fun Credential.toGoogleCredential(): GoogleCredential {
-        return createGoogleCredential(this.accessToken)
-    }
-
     private fun StoredCredential.toGoogleCredential(): GoogleCredential {
         return createGoogleCredential(this.accessToken)
     }
 
     private fun defaultCredential(): GoogleCredential? {
         return flow.credentialDataStore[DATA_STORE_KEY]?.toGoogleCredential()
-    }
-
-    @Throws(IOException::class)
-    fun authorizeUser(): GoogleCredential {
-        if (FtlConstants.useMock) return GoogleCredential()
-
-        val authCode = AuthorizationCodeInstalledApp(flow, LocalServerReceiver())
-        return defaultCredential() ?: authCode.authorize(DATA_STORE_KEY).toGoogleCredential()
     }
 }

